cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/SystemAppModule.java

```diff
@@ -89,19 +94,18 @@ protected void configure() {
     bind(ProgramRuntimeProvider.Mode.class).toInstance(ProgramRuntimeProvider.Mode.LOCAL);
     bind(ProgramRunnerFactory.class).to(DefaultProgramRunnerFactory.class).in(Scopes.SINGLETON);
 
-    bind(MetadataServiceClient.class).to(DefaultMetadataServiceClient.class);
     bind(UGIProvider.class).to(CurrentUGIProvider.class).in(Scopes.SINGLETON);
     bind(AuthenticationContext.class).to(MasterAuthenticationContext.class);
 
     bind(ArtifactRepositoryReader.class).to(RemoteArtifactRepositoryReader.class).in(Scopes.SINGLETON);
-    bind(ArtifactRepository.class).to(DefaultArtifactRepository.class).in(Scopes.SINGLETON);
+    bind(ArtifactRepository.class).to(RemoteArtifactRepository.class).in(Scopes.SINGLETON);
     bind(Impersonator.class).to(DefaultImpersonator.class).in(Scopes.SINGLETON);
```

Comment:
Not needed

Reply:
Did you mean binding the RemoteArtifactRepository or scoping to SINGLETON?

Reply:
I mean the Impersonator binding is not needed. It has the `ImplementedBy` annotation

Reply:
Thanks, removed .